pact/Pact/Core/Errors.hs

```diff
@@ -504,13 +505,26 @@ instance NFData DbOpException
 
 instance Exception DbOpException
 
+data RecoverableError
```

Comment:
Note: This will be replaced in a followup PR very soon. It's already queued up, but I do prefer this PR make it in first